nano/node/vote_cache.cpp

```diff
@@ -244,13 +244,12 @@ std::vector<nano::vote_cache::top_entry> nano::vote_cache::top (const nano::uint
 			cleanup ();
 		}
 
-		for (auto & entry : cache.get<tag_tally> ())
+		for (auto & entry : cache)
 		{
-			if (entry.tally () < min_tally)
+			if (entry.tally () >= min_tally)
 			{
-				break;
+				results.push_back ({ entry.hash (), entry.tally (), entry.final_tally () });
```

Comment:
A std::deque might be a better result container depending on how many items can be returned.

Reply:
> Expansion of a deque is cheaper than the expansion of a [std::vector](https://en.cppreference.com/w/cpp/container/vector) because it does not involve copying of the existing elements to a new memory location.

Interesting, I always assumed deque is using vector under the hood. The result set might in fact be quite large, so using deque here seems like the most reasonable option.

Reply:
However, I'm not sure whether there is any performance penalty when sorting deque? Sorting is performed before returning this result set. I'm planning to add a small benchmarking suite soon that I used for improving performance of stat counting, I'll revisit and measure this once that is done.

Reply:
Deque still has O(1) indexing access so a sort should have minimal impact.